src/adapter/src/coord/message_handler.rs

```diff
@@ -430,6 +433,47 @@ impl Coordinator {
             .await;
     }
 
+    #[tracing::instrument(level = "debug", skip(self, tx, session))]
+    async fn message_create_connection_validation_ready(
+        &mut self,
+        CreateConnectionValidationReady {
+            session,
+            tx,
+            result,
+            params,
+            original_stmt,
+            depends_on,
+            otel_ctx,
+        }: CreateConnectionValidationReady,
+    ) {
+        otel_ctx.attach_as_parent();
+
+        // Ensure that all dependencies still exist after validation, as a
+        // `DROP SECRET` may have sneaked in. If any have gone missing, we
+        // revalidate the original statement. This will either produce a nice
+        // "unknown secret" error, or pick up a new connector that has
```

Comment:
How does this pick up a new connector? Is the ID reused? We document elsewhere we assume they never are: https://github.com/MaterializeInc/materialize/blob/149d2e5bca1f54e94acfe02ab5f5751b9cb2adfc/src/adapter/src/coord.rs#L399

Reply:
This comment is referring to any object dependencies a `CREATE CONNECTION` statement might have and not the connection being created. If any of the dependencies disappear between the initial connection planning and sequencing, which might happen because validation takes a while, then we will replan the statement and try again. During replanning the object names referred by the original `CREATE CONNECTION ..` statement will either resolve to new objects or produce "uknown object X" errors